File: BimManager/BimManager/Entities/Project.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace BimManager.Entities
{
    public class Project
    {

        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public string Description { get; set; }
        [Required]
        public string Status { get; set; }
        [Required]
       
        public string FinishDate { get; set; }
        [Required]

        public decimal Cost { get; set; } = 0;
        [Range(0, 100), Required]
        public int Progress { get; set; } = 0;

        [Required(AllowEmptyStrings = true), DisplayFormat(ConvertEmptyStringToNull = false)]
        public string? IfcName { get; set; }

        [NotMapped]
        public IFormFile IfcFile { get; set; }
        [NotMapped]
        public string IfcSrc { get; set; }
    }
}

File: front/src/class/Project.ts
import {v4 as uuidv4} from 'uuid'
export type ProjectStatus = "Pending" | "Active" | "Finished"

export interface IProject{
    name: string,
    description: string,
    status: ProjectStatus,
    finishDate: Date,
    ifcName: string | null,
    cost: number,
    progress: number,
   
}


export class Project implements IProject{
    name: string = "";
    description: string="";
    status: ProjectStatus = "Active";
    finishDate: Date = new Date();
    ifcName: string | null = null;
    cost: number = 0;
    progress: number = 0;
    id:string = "";

    constructor(data: IProject, id = uuidv4()){
        this.id = id;
        this.name = data.name;
        this.description = data.description;
        this.status = data.status;
        this.finishDate = data.finishDate,
        this.ifcName = data.ifcName,
        this.cost = data.cost;
        this.progress = data.progress;
    }
}